packetbeat/protos/cassandra/internal/gocql/frame.go

```diff
@@ -168,11 +168,18 @@ func (f *Framer) ReadHeader() (head *frameHeader, err error) {
 // reads a frame form the wire into the framers buffer
 func (f *Framer) ReadFrame() (data map[string]interface{}, err error) {
 	defer func() {
-		if r := recover(); r != nil {
-			if _, ok := r.(runtime.Error); ok {
-				panic(r)
-			}
-			err = r.(error)
+		// The casandra plugin uses panic for flow control, panicking on
```

Comment:
> The casandra plugin uses panic for flow control

😮

Reply:
Same in encoding/json for the marshalling code. https://github.com/golang/go/blob/518889b35cb07f3e71963f2ccfc0f96ee26a51ce/src/encoding/json/encode.go#L315-L337